```
Error running mint automation
+ mkdir -p 14391-cf936ce/gopath/src/github.com/minio
+ git -C 14391-cf936ce/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 14391-cf936ce/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 14391-cf936ce/gopath/src/github.com/minio/minio fetch --quiet minio pull/14391/head:pr14391
+ git -C 14391-cf936ce/gopath/src/github.com/minio/minio checkout --quiet pr14391
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/14391-cf936ce/gopath
+ make -C 14391-cf936ce/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/pkg v1.1.15
go: downloading cloud.google.com/go/storage v1.10.0
go: downloading github.com/minio/madmin-go v1.3.1
go: downloading google.golang.org/api v0.58.0
go: downloading github.com/minio/minio-go/v7 v7.0.22
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/fatih/color v1.13.0
go: downloading github.com/colinmarc/hdfs/v2 v2.2.0
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/klauspost/compress v1.13.6
go: downloading github.com/minio/sio v0.3.0
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading golang.org/x/crypto v0.0.0-20220128200615-198e4374d7ed
go: downloading github.com/Shopify/sarama v1.27.2
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/golang-jwt/jwt/v4 v4.1.0
go: downloading golang.org/x/sys v0.0.0-20220128215802-99c3d69c2c27
go: downloading github.com/buger/jsonparser v1.1.1
go: downloading github.com/rs/dnscache v0.0.0-20211102005908-e0241e321417
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/minio/kes v0.14.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/bits-and-blooms/bloom/v3 v3.0.1
go: downloading github.com/go-openapi/loads v0.21.0
go: downloading github.com/minio/console v0.14.7
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/inconshreveable/mousetrap v1.0.0
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading github.com/minio/dperf v0.2.1-0.20220126071331-eefa6468e289
go: downloading github.com/coreos/go-oidc v2.1.0+incompatible
go: downloading github.com/cosnicolaou/pbzip2 v1.0.1
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/miekg/dns v1.1.43
go: downloading github.com/klauspost/readahead v1.4.0
go: downloading github.com/felixge/fgprof v0.9.2
go: downloading github.com/klauspost/reedsolomon v1.9.15
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/minio/selfupdate v0.4.0
go: downloading github.com/minio/zipindex v0.2.1
go: downloading github.com/philhofer/fwd v1.1.2-0.20210722190033-5c56ac6d0bb9
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/pierrec/lz4 v2.6.0+incompatible
go: downloading github.com/prometheus/client_golang v1.11.0
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/prometheus/procfs v0.7.3
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/shirou/gopsutil/v3 v3.21.12
go: downloading github.com/tinylib/msgp v1.1.7-0.20211026165309-e818a1881b0e
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading github.com/yargevad/filepathx v1.0.0
go: downloading github.com/zeebo/xxh3 v1.0.0
go: downloading go.etcd.io/etcd/api/v3 v3.5.1
go: downloading go.etcd.io/etcd/client/v3 v3.5.1
go: downloading go.uber.org/atomic v1.9.0
go: downloading golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f
go: downloading github.com/beevik/ntp v0.3.0
go: downloading github.com/go-ldap/ldap/v3 v3.2.4
go: downloading github.com/coredns/coredns v1.4.0
go: downloading golang.org/x/time v0.0.0-20211116232009-f0f3c7e86c11
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.5
go: downloading github.com/elastic/go-elasticsearch/v7 v7.12.0
go: downloading github.com/go-sql-driver/mysql v1.6.0
go: downloading github.com/gomodule/redigo v1.8.8
go: downloading github.com/lib/pq v1.10.4
go: downloading github.com/nats-io/nats.go v1.13.1-0.20220121202836-972a071d373d
go: downloading github.com/nats-io/stan.go v0.10.2
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/streadway/amqp v1.0.0
go: downloading github.com/montanaflynn/stats v0.6.6
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/golang-jwt/jwt v3.2.2+incompatible
go: downloading go.uber.org/zap v1.19.1
go: downloading github.com/minio/simdjson-go v0.2.1
go: downloading github.com/mattn/go-colorable v0.1.12
go: downloading github.com/mattn/go-isatty v0.0.14
go: downloading golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
go: downloading github.com/pquerna/cachecontrol v0.0.0-20171018203845-0dec1b30a021
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: downloading github.com/bits-and-blooms/bitset v1.2.0
go: downloading github.com/mattn/go-runewidth v0.0.13
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/klauspost/cpuid/v2 v2.0.9
go: downloading golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd
go: downloading github.com/minio/argon2 v1.0.0
go: downloading github.com/minio/csvparser v1.0.0
go: downloading github.com/minio/parquet-go v1.1.0
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.1
go: downloading github.com/golang/snappy v0.0.3
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/go-openapi/analysis v0.21.2
go: downloading github.com/go-openapi/spec v0.20.4
go: downloading github.com/go-openapi/swag v0.19.15
go: downloading github.com/prometheus/common v0.32.1
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/google/pprof v0.0.0-20211214055906-6f57359322fd
go: downloading go.etcd.io/etcd/client/pkg/v3 v3.5.1
go: downloading google.golang.org/grpc v1.43.0
go: downloading github.com/rivo/uniseg v0.2.0
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/lestrrat-go/jwx v1.2.14
go: downloading github.com/fatih/structs v1.1.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading google.golang.org/genproto v0.0.0-20211223182754-3ac035c7e7cb
go: downloading golang.org/x/text v0.3.7
go: downloading cloud.google.com/go v0.94.1
go: downloading go.uber.org/multierr v1.7.0
go: downloading github.com/rs/xid v1.3.0
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading gopkg.in/ini.v1 v1.66.3
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading github.com/minio/md5-simd v1.1.2
go: downloading github.com/nats-io/nkeys v0.3.0
go: downloading github.com/mailru/easyjson v0.7.6
go: downloading google.golang.org/protobuf v1.27.1
go: downloading github.com/tklauser/go-sysconf v0.3.9
go: downloading github.com/apache/thrift v0.15.0
go: downloading github.com/coreos/go-systemd/v22 v22.3.2
go: downloading github.com/tidwall/gjson v1.12.1
go: downloading github.com/tidwall/sjson v1.2.3
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.5.0
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/tidwall/match v1.1.1
go: downloading github.com/tidwall/pretty v1.2.0
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/tklauser/numcpus v0.3.0
go: downloading github.com/lestrrat-go/backoff/v2 v2.0.8
go: downloading github.com/lestrrat-go/option v1.0.0
go: downloading github.com/lestrrat-go/iter v1.0.1
go: downloading github.com/lestrrat-go/httpcc v1.0.0
go: downloading github.com/lestrrat-go/blackmagic v1.0.0
go: downloading github.com/go-openapi/strfmt v0.21.1
go: downloading github.com/go-openapi/jsonpointer v0.19.5
go: downloading github.com/go-openapi/jsonreference v0.19.6
go: downloading github.com/oklog/ulid v1.3.1
go: downloading github.com/go-openapi/errors v0.20.2
go: downloading go.mongodb.org/mongo-driver v1.7.5
go: downloading github.com/asaskevich/govalidator v0.0.0-20200907205600-7a23bdc65eef
go: downloading github.com/mitchellh/mapstructure v1.4.2
go: downloading github.com/PuerkitoBio/purell v1.1.1
go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
go: downloading github.com/googleapis/gax-go/v2 v2.1.1
go: downloading go.opencensus.io v0.23.0
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading github.com/go-stack/stack v1.8.0
go: downloading github.com/go-openapi/runtime v0.21.1
go: downloading github.com/minio/mc v0.0.0-20220204044644-e048c85d71a7
go: downloading github.com/unrolled/secure v1.0.9
go: downloading github.com/minio/direct-csi v1.3.5-0.20210601185811-f7776f7961bf
go: downloading github.com/jessevdk/go-flags v1.4.0
go: downloading k8s.io/client-go v0.21.1
go: downloading github.com/minio/operator v0.0.0-20220110040724-a5d59a342b7f
go: downloading github.com/go-openapi/validate v0.20.3
go: downloading github.com/docker/go-units v0.4.0
go: downloading k8s.io/apimachinery v0.21.1
go: downloading github.com/mb0/glob v0.0.0-20160210091149-1eb79d2de6c4
go: downloading k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
go: downloading github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
go: downloading github.com/charmbracelet/lipgloss v0.4.1-0.20220204041308-bf2912e703f6
go: downloading github.com/charmbracelet/bubbles v0.10.0
go: downloading github.com/charmbracelet/bubbletea v0.19.3
go: downloading github.com/minio/colorjson v1.0.1
go: downloading github.com/pkg/xattr v0.4.4
go: downloading github.com/posener/complete v1.2.3
go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
go: downloading github.com/pkg/profile v1.6.0
go: downloading github.com/minio/filepath v1.0.0
go: downloading maze.io/x/duration v0.0.0-20160924141736-faac084b6075
go: downloading github.com/olekukonko/tablewriter v0.0.5
go: downloading gopkg.in/h2non/filetype.v1 v1.0.5
go: downloading golang.org/x/term v0.0.0-20210927222741-03fcf44c2211
go: downloading github.com/muesli/termenv v0.11.1-0.20220204035834-5ac8409525e0
go: downloading github.com/muesli/reflow v0.3.0
go: downloading github.com/lucasb-eyer/go-colorful v1.2.0
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/containerd/console v1.0.2
go: downloading github.com/muesli/ansi v0.0.0-20211018074035-2e021307bc4b
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading k8s.io/klog/v2 v2.30.0
go: downloading k8s.io/api v0.21.1
go: downloading sigs.k8s.io/controller-runtime v0.8.0
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.1.0
go: downloading sigs.k8s.io/yaml v1.2.0
go: downloading github.com/google/gofuzz v1.1.0
go: downloading github.com/google/go-cmp v0.5.6
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/go-logr/logr v1.2.0
go: downloading github.com/emicklei/go-restful v2.9.5+incompatible
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading k8s.io/utils v0.0.0-20201110183641-67b214c5f920
go: downloading github.com/googleapis/gnostic v0.5.1
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
# github.com/minio/minio/cmd
cmd/site-replication.go:23:2: imported and not used: "crypto/tls"
cmd/site-replication.go:29:2: imported and not used: "net/http"
cmd/site-replication.go:2063:14: undefined: newRemoteClusterHTTPTransport
make: *** [build] Error 2

```

--- a/cmd/site-replication.go
+++ b/cmd/site-replication.go
@@ -20,13 +20,11 @@ package cmd
 import (
 	"bytes"
 	"context"
-	"crypto/tls"
 	"encoding/base64"
 	"encoding/json"
 	"encoding/xml"
 	"errors"
 	"fmt"
-	"net/http"
 	"net/url"
 	"reflect"
 	"sort"
@@ -2040,26 +2038,13 @@ func (c *SiteReplicationSys) RemoveRemoteTargetsForEndpoint(ctx context.Context,
 
 // Other helpers
 
-// newRemoteClusterHTTPTransport returns a new http configuration
-// used while communicating with the remote cluster.
-func newRemoteClusterHTTPTransport() *http.Transport {
-	tr := &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
-		TLSClientConfig: &tls.Config{
-			RootCAs:            globalRootCAs,
-			ClientSessionCache: tls.NewLRUClientSessionCache(tlsClientSessionCacheSize),
-		},
-	}
-	return tr
-}
-
 func getAdminClient(endpoint, accessKey, secretKey string) (*madmin.AdminClient, error) {
 	epURL, _ := url.Parse(endpoint)
 	client, err := madmin.New(epURL.Host, accessKey, secretKey, epURL.Scheme == "https")
 	if err != nil {
 		return nil, err
 	}
-	client.SetCustomTransport(newRemoteClusterHTTPTransport())
+	client.SetCustomTransport(NewRemoteTargetHTTPTransport())
 	return client, nil
 }
 
@@ -2071,7 +2056,7 @@ func getS3Client(pc madmin.PeerSite) (*minioClient.Client, error) {
 	return minioClient.New(ep.Host, &minioClient.Options{
 		Creds:     credentials.NewStaticV4(pc.AccessKey, pc.SecretKey, ""),
 		Secure:    ep.Scheme == "https",
-		Transport: newRemoteClusterHTTPTransport(),
+		Transport: NewRemoteTargetHTTPTransport(),
 	})
 }
 

--- a/cmd/warm-backend-s3.go
+++ b/cmd/warm-backend-s3.go
@@ -21,14 +21,22 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"net/http"
 	"net/url"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/minio/madmin-go"
 	minio "github.com/minio/minio-go/v7"
 	miniogo "github.com/minio/minio-go/v7"
 	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+// getRemoteTierTargetInstanceTransport contains a singleton roundtripper.
+var (
+	getRemoteTierTargetInstanceTransport     http.RoundTripper
+	getRemoteTierTargetInstanceTransportOnce sync.Once
 )
 
 type warmBackendS3 struct {
@@ -109,13 +117,13 @@ func newWarmBackendS3(conf madmin.TierS3) (*warmBackendS3, error) {
 	} else {
 		creds = credentials.NewStaticV4(conf.AccessKey, conf.SecretKey, "")
 	}
-	getRemoteTargetInstanceTransportOnce.Do(func() {
-		getRemoteTargetInstanceTransport = newGatewayHTTPTransport(10 * time.Minute)
+	getRemoteTierTargetInstanceTransportOnce.Do(func() {
+		getRemoteTierTargetInstanceTransport = newGatewayHTTPTransport(10 * time.Minute)
 	})
 	opts := &minio.Options{
 		Creds:     creds,
 		Secure:    u.Scheme == "https",
-		Transport: getRemoteTargetInstanceTransport,
+		Transport: getRemoteTierTargetInstanceTransport,
 	}
 	client, err := minio.New(u.Host, opts)
 	if err != nil {

